src/upasana/Shop.java

```diff
@@ -0,0 +1,133 @@
+package upasana;
+
+public class Shop {
+	int maggieTQty;
+	int dosaTQty;
+	int pouchestQty;
+	int panipuriTQty;
+	int masalaTQty;
+	
+	
+	public void setQty(int maggieReqQty,int dosaReqQty,int pouchesReqQty,int panipuriReqQty,int masalaReqQty) {
+		maggieTQty = maggieReqQty;
+		dosaTQty = dosaReqQty;
+		pouchestQty = pouchesReqQty;
+		panipuriTQty =  panipuriReqQty;
+		masalaTQty= masalaReqQty;
+		
+	}
+	
+	public void purchaseItems(int maggieReqQty,int dosaReqQty,int pouchesReqQty,int panipuriReqQty,int masalaReqQty,boolean pSAllowed) {
+		
+		if(maggieReqQty<=maggieTQty) {
+			maggieTQty = maggieTQty - maggieReqQty;
+			System.out.println("Available qty of maggie is "+maggieTQty + "  and sold qty of maggie is "+maggieReqQty);
+		}	
+		else {
+				if(pSAllowed) {
+					maggieTQty=0;
```

Comment:
If request stock is not available mean we are setting direct it zero, 
what if second time your calling same method and let say you request 20 packs, then this looks not logically correct